src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -37,7 +46,106 @@ public static IEnumerable<ApplicationPart> DiscoverAssemblyParts(string entryPoi
             var entryAssembly = Assembly.Load(new AssemblyName(entryPointAssemblyName));
             var context = DependencyContext.Load(entryAssembly);
 
-            return GetCandidateAssemblies(entryAssembly, context).Select(p => new AssemblyPart(p));
+            var candidateAssemblies = GetCandidateAssemblies(entryAssembly, context);
+            var additionalReferences = candidateAssemblies
+                .ToDictionary(
```

Comment:
Does this really need to be a dictionary? It seems like this is just a select, you iterate the results instead of indexing them.